src/PayU/PaymentsApi/AluV3/Services/ResponseBuilder.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+
+namespace PayU\PaymentsApi\AluV3\Services;
+
+use PayU\Alu\Exceptions\ClientException;
+use PayU\Alu\Response;
+use PayU\Alu\ResponseWireAccount;
+use PayU\PaymentsApi\AluV3\Entities\AuthorizationResponse;
+use PayU\PaymentsApi\AluV3\Exceptions\ResponseBuilderException;
+use SimpleXMLElement;
+
+final class ResponseBuilder
+{
+
+    /**
+     * @param AuthorizationResponse $authorizationResponse
+     * @param HashService $hashService
+     * @return Response
+     * @throws ResponseBuilderException
+     */
+    public function buildResponse(AuthorizationResponse $authorizationResponse, HashService $hashService)
+    {
+        $response = $this->build($authorizationResponse);
+        if ('' != $response->getHash()) {
```

Comment:
Is it possible to receive a response without a hash?

Reply:
Yes. From our docs https://secure.payu.ro/docs/alu/v3/:
> If the signature is NOT correct you'll get a HASH_MISMATCH error and the HASH tag will be empty.